### PR TITLE
Add support for `controller.selectorLabels`

### PR DESCRIPTION
**Pull request by hanyouqing**
### Proposed changes
Fix issue https://github.com/nginxinc/kubernetes-ingress/issues/3976

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Comment by jasonwilliams14** [comment]
@hanyouqing Thank you for the PR. We are currently reviewing this on our side and trying to reproduce the issue. If you could respond to our request for additional information in the issue thread you created, that would be great. Thank you.

**Comment by hanyouqing** [comment]
> @hanyouqing Thank you for the PR. We are currently reviewing this on our side and trying to reproduce the issue. If you could respond to our request for additional information in the issue thread you created, that would be great. Thank you.

Thanks for your reply! I'm glad to share the reproduce steps, but I can't reproduce it with Helm. We manage the helm chart with Terraform, It might be some issues for the helm Terraform provider. Let me figure it out, if so I'll close the pr.

**Comment by hanyouqing** [comment]
Hi @jasonwilliams14, I've added some comments and how to reproduce it see the issue page https://github.com/nginxinc/kubernetes-ingress/issues/3976#issuecomment-1612516304. 
Thanks!

File: deployments/helm-chart/templates/_helpers.tpl [hunk 1]
```diff
@@ -64,9 +64,13 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
...
 */}}
 {{- define "nginx-ingress.selectorLabels" -}}
+{{- if .Values.controller.selectorLabels }}
+{{ toYaml .Values.controller.selectorLabels }}
+{{- else -}}
 app.kubernetes.io/name: {{ include "nginx-ingress.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
-{{- end }}
+{{- end -}}
+{{- end -}}
```
Comment (lucacome) [inline deployments/helm-chart/templates/_helpers.tpl:73]:
```suggestion
{{- end }}
```

**Review by lucacome — APPROVED** [review]
Looks good @hanyouqing

**Review by vepatel — APPROVED** [review]

File: deployments/helm-chart/templates/_helpers.tpl [hunk 2]
```diff
@@ -64,9 +64,13 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
 Selector labels
 */}}
 {{- define "nginx-ingress.selectorLabels" -}}
+{{- if .Values.controller.selectorLabels }}
```
Comment (lucacome) [inline deployments/helm-chart/templates/_helpers.tpl:67]:
```suggestion
{{- if .Values.controller.selectorLabels -}}
```

**Pull request merged**